src/Network/Network/LoadBalancer/GetAzureLoadBalancerBackendPool.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Linq;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "LoadBalancerBackendAddressPool"), OutputType(typeof(PSBackendAddressPool))]
+    public partial class GetAzureLoadBalancerBackendPool : NetworkBaseCmdlet
+    {
+
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+        private const string GetByParentObjectParameterSet = "GetByParentObjectParameterSet";
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group name of the load balancer.", ParameterSetName = GetByNameParameterSet)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the load balancer.", ParameterSetName = GetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string LoadBalancerName { get; set; }
+
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the backend address pool.", ParameterSetName = GetByNameParameterSet)]
+        [Parameter(Mandatory = false, HelpMessage = "The name of the backend address pool.", ParameterSetName = GetByParentObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Name {get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = GetByParentObjectParameterSet)]
+        [ValidateNotNull]
+        public PSLoadBalancer LoadBalancer { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = GetByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+
+        public override void Execute()
+        {
+            base.Execute();
+            BackendAddressPool loadBalancerBackendAddressPool = null;
+
+
+            if (this.IsParameterBound(c => c.LoadBalancer))
+            {
+                this.ResourceGroupName = this.LoadBalancer.ResourceGroupName;
+                this.LoadBalancerName = this.LoadBalancer.Name;
+            }
+
+            if (this.IsParameterBound(p => p.ResourceId))
+            {
+                ResourceIdentifier resourceIdentifier = new ResourceIdentifier(ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.LoadBalancerName = resourceIdentifier.ParentResource.Split('/')[1];
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+
+            // Get single backend pool
+            if (ShouldGetByName(this.ResourceGroupName, this.Name))
```

Comment:
where is this shouldgetbyname method defined?

Reply:
Microsoft.Azure.Commands.ResourceManager.Common , its a common thing across all the existing cmdlets to use that method before calling by RG and resource name.